homeassistant/components/intent/__init__.py

```diff
@@ -70,16 +72,23 @@ async def async_call_service(self, intent_obj: intent.Intent, state: State) -> N
         if state.domain == COVER_DOMAIN:
             # on = open
             # off = close
-            await hass.services.async_call(
-                COVER_DOMAIN,
-                SERVICE_OPEN_COVER
-                if self.service == SERVICE_TURN_ON
-                else SERVICE_CLOSE_COVER,
-                {ATTR_ENTITY_ID: state.entity_id},
-                context=intent_obj.context,
-                blocking=True,
-                limit=self.service_timeout,
+            task = hass.async_create_task(
+                hass.services.async_call(
+                    COVER_DOMAIN,
+                    SERVICE_OPEN_COVER
+                    if self.service == SERVICE_TURN_ON
+                    else SERVICE_CLOSE_COVER,
+                    {ATTR_ENTITY_ID: state.entity_id},
+                    context=intent_obj.context,
+                    blocking=True,
+                )
             )
+            # Block with a short timeout to (hopefully) catch validation errors
+            try:
+                async with async_timeout.timeout(self.service_timeout):
+                    await task
+            except asyncio.TimeoutError:
+                pass  # Expected
```

Comment:
Was there supposed to be a return here somewhere? It looks like it falls through to call `await super().async_call_service(intent_obj, state)` below and run the service call again.

Reply:
Yes, we should return here, or add an `else` on line 98.